api/src/server/auth.rs

```diff
@@ -70,13 +85,18 @@ pub async fn build_client_map(open_id_config: &OpenIdConfig) -> ClientMap {
 pub async fn route_authorize(
     client_map: web::Data<ClientMap>,
     params: web::Path<(String,)>,
+    query: web::Query<RedirectQuery>,
 ) -> Result<HttpResponse, actix_web::Error> {
-    // TODO: handle bad name
-    let provider_name = params.0.to_string();
-    let oidc_client = client_map.get_client(&provider_name)?;
+    let provider_name: &str = &params.0;
+    let oidc_client = client_map.get_client(provider_name)?;
+    let state = AuthState {
+        next: query.next.as_deref(),
+    };
 
     let auth_url = oidc_client.auth_url(&Options {
         scope: Some("email".into()),
+        // Serialization shouldn't ever fail so yeet that shit outta the Result
```

Comment:
This technically could fail if they just hit this endpoint manually but idk if that matters. Does `actix_web` kill the server on panic?

Reply:
I also ran into issues before where some (bad) openid providers would mess with the formatting of the state paraem but idk if we should care about that either

Reply:
Here's what the docs say about when `serde_json::to_string` can fail:

> Serialization can fail if T's implementation of Serialize decides to fail, or if T contains a map with non-string keys.

Since `T` is `AuthState` which is just a simple derived serializer, it can never fail. And we don't have any maps in here, so the second part can't happen either. So `serde_json::to_string` can literally never fail in this case. But yeah if there is a panic, actix kills that thread (but i think the rest of the server keeps running).